deploy-deps.sh

```diff
@@ -86,7 +86,7 @@ deploy_tekton() {
 
 deploy_cert_manager() {
     kubectl apply -k "${script_path}/dependencies/cert-manager"
-    retry "kubectl wait --for=condition=Ready --timeout=120s -l app.kubernetes.io/instance=cert-manager -n cert-manager pod" \
+    retry "kubectl wait --for=condition=Ready --timeout=180s -l app.kubernetes.io/instance=cert-manager -n cert-manager pod" \
```

Comment:
(I don't have a good solution to this issue)

Reply:
This will apply to any retry we do. Not sure that we want that. It might be that, instead, we can sleep a little before this retry to reduce the chance of this happening. This should not cost us anything if we assume that the retry will either iterate more than once or the pod will take some time to be ready.

Reply:
We can use a parameter for how many retries we want, it involves changes to the retry and all the uses of it.
Or as you mentioned we can add sleep before the retry, something like:
```
retry() {
    sleep 3
    for _ in {1..3}; do
        local ret=0
        $1 || ret="$?"
        if [[ "$ret" -eq 0 ]]; then
            return 0
        fi
        sleep 3
    done

    echo "$1": "$2."
    return "$ret"
}
```

Reply:
I meant adding a sleep before the retry is called specifically for the case of cert-manger. I don't think it will have any cost in terms of execution time for the entire script - it will just move the waiting for an earlier stage.

Yes. We can also introduce a new arg to the retry function as you mentioned.

Reply:
Done (adding sleep before the retry)